com.unity.render-pipelines.universal/Runtime/ScriptableRenderer.cs

```diff
@@ -724,7 +770,62 @@ void ExecuteRenderPass(ScriptableRenderContext context, ScriptableRenderPass ren
             context.ExecuteCommandBuffer(cmd);
             CommandBufferPool.Release(cmd);
 
-            renderPass.Execute(context, ref renderingData);
+            if (IsRenderPassEnabled(renderPass) && cameraData.cameraType == CameraType.Game)
+            {
+                int validColorBuffersCount = (int)RenderingUtils.GetValidColorBufferCount(renderPass.colorAttachments);
+
+                bool isLastPass = renderPass.srpDescriptor.isLastPass;
+                bool isLastPassToBB = isLastPass && (m_ActiveColorAttachmentDescriptors[0].loadStoreTarget == BuiltinRenderTextureType.CameraTarget);
+                bool useDepth = m_ActiveDepthAttachment == RenderTargetHandle.CameraTarget.Identifier() && (!(isLastPassToBB || (isLastPass && cameraData.camera.targetTexture != null)));
+
+                var attachments =
+                    new NativeArray<AttachmentDescriptor>(useDepth && !renderPass.srpDescriptor.isDepthOnly ? validColorBuffersCount + 1 : 1, Allocator.Temp);
+
+                for (int i = 0; i < validColorBuffersCount; ++i)
+                    attachments[i] = m_ActiveColorAttachmentDescriptors[i];
+
+                if (useDepth && !renderPass.srpDescriptor.isDepthOnly)
+                    attachments[validColorBuffersCount] = m_ActiveDepthAttachmentDescriptor;
```

Comment:
This could be encapsulated in the ScriptableRenderPass when configuring attachments we could already cache attachment descriptors?

Reply:
This is temporary code as it will be changing in the future with lazy merging being introduced. We technically could cache something per-pass, but that will change as soon as we start grouping the render passes